### ISSUE TITLE
Public Logging of Object Request Rejections

### ISSUE THREAD

**Issue by martinhannigan** [peeringdb/peeringdb#1408]
**Is your feature request related to a problem? Please describe.**

PeeringDB is light on actual policy, but the policy we do have can be supported by transparency. We have cases where policy appears to not be followed and where there is no policy but could be better products or implementation. Data is key.

**Who is affected by the problem?**

PDB users and supporters at large who desire to help drive valid issues.

**What is the impact?**

Transparency

**Are there security concerns?**

Doubtful

**Are there privacy concerns?**

Perhaps, but the request  output can be masked adequately and automatically to protect privacy if it applies.

**Describe the solution you'd like**

An automated and public log of all rejected objects requests; a sub log or other easy indication of when an object rejection had been appealed and forwarded per our existing policy.

**Do you think this feature will require a formal design?**

Uncertain.

**Describe alternatives you've considered**

Doing nothing. This would leave these types of actions in the blackhole they're already in and erode confidence in PDB.

**Could this feature request need support from the Admin Committee?**
No.

**What is the proposed priority?**

Medium

**Provide a rationale for any/all of the above**

There's limited view outside the AC of what's being rejected. Or statistics by who. And if a rejected request is appealed and how often. We have no visibility into timing of the overall process. With data, the community and volunteers can provide better oversight. Benefits for clear product improvements or better understanding of why a rejection makes sense for one type of request, but not another which is better handled transparently, for example.

**Additional context**

Thanks!

**Comment by peterhelmenstine** [comment]
+1 I'm all for transparency

**Comment by grizz** [comment]
I think this makes sense but am a bit worried about the actual data being displayed.

+1 if it's just the object type and reason -- more than that I would want to see more discussion.

**Comment by arnoldnipper** [comment]
Actually, afaik @peeringdb/ac doesn't reject addition of any objects. At least, I don't but ask for clarifation. In 95%, I don't get an answer, and the request is cleared after 90 days.

As much as I'm for transparency, looking at the pile of issues still waiting, I would queue this issue at the end

**Comment by martinhannigan** [comment]
Agreed. It would be good to take a look at it how its laid out and see if
we can filter it but leave it intact to make sense. Not entirely certain
that's possible.

**Comment by leovegoda** [comment]
> Agreed. It would be good to take a look at it how its laid out and see if we can filter it but leave it intact to make sense. Not entirely certain that's possible.

Were we to do this, should the log go on docs.peeringdb.com or www.peeringdb.com?

**Comment by martinhannigan** [comment]
docs would make sense to me. Perhaps another menu item for "logs" and just
make it another matter-of-fact thing we do.

**Comment by peterhelmenstine** [comment]
Actually I second Arnold's point. I don't think I've ever actually denied any but said "hey you don't look like a facility. Come back with more details" and we never get a response. Except for all those Brazilian mobile sales huts in the middle of nowhere. I reject those.

**Comment by martinhannigan** [comment]
Good point and incredibly insightful as well as a bit shocking.

How many answered which were not rejected, but we’re not approved? How many had successfully created objects before but were’ “asked” and responded, but were not created without going to the board?



From: Peter Helmenstine ***@***.***>
Date: Friday, July 7, 2023 at 7:03 PM
To: peeringdb/peeringdb ***@***.***>
Cc: Martin Hannigan ***@***.***>, Assign ***@***.***>
Subject: Re: [peeringdb/peeringdb] Public Logging of Object Request Rejections (Issue #1408)

Actually I second Arnold's point. I don't think I've ever actually denied any but said "hey you don't look like a facility. Come back with more details" and we never get a response. Except for all those Brazilian mobile sales huts in the middle of nowhere. I reject those.

**Comment by peterhelmenstine** [comment]
I honestly can only recall maybe a handful (including yours) that has been debated in the 5 years I've been looking at tickets.

**Comment by martinhannigan** [comment]
Thanks Peter!

I’ve never seen the ticket queue or what’s in it so I wouldn’t know exactly what would be best. This process is very useful to sharpen issues before consensus so thank you for that. I would suggest starting with rejection, what was requested for “another set of eyes” “what went to trustees” board, or something that would reflect that we should a) automate to reduce workflow wherever possible and make only exceptions into tickets or b) investigate patterns of concern. This seems reasonable, more so than ‘we don’t actually do that but we do it this way…’

I didn’t know my requests were part of only a handful. That count was > 4. That doesn’t include the FAC object which was hijacked. All of the records which were hung up for weeks on end in a queue are in the database. Thanks to the intervention of the board. That’s not what motivated my issue, but I appreciate adding it.

Warm regards,

-M<

From: Peter Helmenstine ***@***.***>
Date: Friday, July 7, 2023 at 7:41 PM
To: peeringdb/peeringdb ***@***.***>
Cc: Martin Hannigan ***@***.***>, Assign ***@***.***>
Subject: Re: [peeringdb/peeringdb] Public Logging of Object Request Rejections (Issue #1408)

I honestly can only recall maybe a handful (including yours) that has been debated in the 5 years I've been looking at tickets.

**Comment by grizz** [comment]
Per PC call, punting discussion to a later date.

**Comment by mcmanuss8** [comment]
+1 to adding transparency, but I think there's not a completely straightforward technical solution here from my understanding of the database structure. I propose we have 20C propose a solution here that might fit the need and then vote on if it's sufficient. 

If we do this I'd also suggest we revisit #581 which we closed because we didn't have a strong enough use case - well this is a use case where we could put accepts, rejects and rejects w/ escalation/appeal in for the public to see, which I think would be valuable

**Comment by martinhannigan** [comment]
+1

**Comment by leovegoda** [comment]
In [yesterday's meeting](https://docs.peeringdb.com/committee/product/notes/2023-11-02_Product_Committee_Notes.pdf) we agreed to investigate what data we have, so we can work out what could be automated

**Comment by martinhannigan** [comment]
We also seemed to have agreed that high level data that establishes
patterns is better than full disclosure - from a cost and effort
perspective.

Thanks!

-M<

**Comment by martinhannigan** [comment]
We have consensus to do this. How do we move it ahead?

**Comment by leovegoda** [comment]
> We have consensus to do this. How do we move it ahead?

The @peeringdb/pc will review "what information we have in DeskPro about rejections: what data is available in structured formats? Are there patterns we should be aware of?" at the December meeting. If we can use whate we already have then that is great. If we can't we might need to make improvements to that data first.

**Comment by martinhannigan** [comment]
Perhaps there is a better, wider, way to accomplish some security around our requests being handled in a fair and even manner which could make this easier? First is a security policy defining who has access to what files e.g. logs, search history, etc. Are my searches and/or API uses secure? I honestly don't know.

Second is an automation policy which requires auto approval for object creation. Automation as policy is a good way to improve fairness. In the automated world it's hard to make a subjective decision which results in more even application of policy. The higher the automation the better the fairness IMHO.

The trick is to automate to 95% instead of 5%. 

Let's discuss again at upcoming PC meeting. Maybe we can figure out some less hard more easily applied methods?

**Comment by peterhelmenstine** [comment]
I believe this was pending me to get actual examples of rejections from the AC however that's proving to be more difficult than I thought. Having a hard time finding them because there aren't many. A majority are kicked back asking for info and then getting nothing back. Not a formal rejection. The only example I can recall would be those mobile sales offices in the jungle that are not facilities we get regularly for some reason. And Marty's site without a website. I can't find any others.

**Comment by martinhannigan** [comment]
It was. We talked about the mobile sales office example. I would argue
ignoring without rejecting  (or waitint for a time out to close a ticket)
is akin to silent rejection. The AC isn't transparent enough to ensure
everyone is being treated fairly. There has to be  simple waya to
demonstrate that, including improving upon automation and ensuring that
steps requiring AC ``Approval" aren't unnecessary ways to control the
platform and apply subjective criteria one way or the other.

Thanks for checking.
